model/simulationtests/fan_componentmodel.rb

```diff
@@ -0,0 +1,308 @@
+vFDPartLoad.setCoefficient2C2(0.0155361)
+vFDPartLoad.setCoefficient3C3(-0.0059365)
+vFDPartLoad.setMinimumValueofx(0.0)
+vFDPartLoad.setMaximumValueofx(1.0)
+vFDPartLoad.setMinimumCurveOutput(0.01)
+vFDPartLoad.setMaximumCurveOutput(1.0)
+# vFDPartLoad.setInputUnitTypeforx("")
+# vFDPartLoad.setOutputUnitType("")
+
+fan.setName('Fan Component Model')
+fan.autosizeMaximumFlowRate
+fan.autosizeMinimumFlowRate
+fan.setFanSizingFactor(1.0)
+fan.setFanWheelDiameter(0.3048)
+fan.setFanOutletArea(0.0873288576)
+fan.setMaximumFanStaticEfficiency(0.514)
+fan.setEulerNumberatMaximumFanStaticEfficiency(9.76)
+fan.setMaximumDimensionlessFanAirflow(0.160331811647483)
+fan.autosizeMotorFanPulleyRatio
+fan.autosizeBeltMaximumTorque
+fan.setBeltSizingFactor(1.0)
+fan.setBeltFractionalTorqueTransition(0.167)
+fan.setMotorMaximumSpeed(1800.0)
+fan.autosizeMaximumMotorOutputPower
+fan.setMotorSizingFactor(1.0)
+fan.setMotorInAirstreamFraction(1.0)
+fan.setVFDEfficiencyType('Power')
+fan.autosizeMaximumVFDOutputPower
+fan.setVFDSizingFactor(1.0)
+fan.setFanPressureRiseCurve(vSDExample)
+fan.setDuctStaticPressureResetCurve(diagnosticSPR)
+fan.setNormalizedFanStaticEfficiencyCurveNonStallRegion(fanEff120CPLANormal)
+fan.setNormalizedFanStaticEfficiencyCurveStallRegion(fanEff120CPLAStall)
+fan.setNormalizedDimensionlessAirflowCurveNonStallRegion(fanDimFlowNormal)
+fan.setNormalizedDimensionlessAirflowCurveStallRegion(fanDimFlowStall)
+fan.setMaximumBeltEfficiencyCurve(beltMaxEffMedium)
+fan.setNormalizedBeltEfficiencyCurveRegion1(beltPartLoadRegion1)
+fan.setNormalizedBeltEfficiencyCurveRegion2(beltPartLoadRegion2)
+fan.setNormalizedBeltEfficiencyCurveRegion3(beltPartLoadRegion3)
+fan.setMaximumMotorEfficiencyCurve(motorMaxEffAvg)
+fan.setNormalizedMotorEfficiencyCurve(motorPartLoad)
+fan.setVFDEfficiencyCurve(vFDPartLoad)
+
+# Add this fan as the supply fan of a regular AirLoopHVAC, removing the
+# existing FanVariableVolume
+regularAirLoopHVAC.supplyFan.get.remove
+fan.addToNode(regularAirLoopHVAC.supplyOutletNode)
+
+# AirLoopHVACUnitarySystem
+unitary_system = OpenStudio::Model::AirLoopHVACUnitarySystem.new(model)
+unitary_system_fan = OpenStudio::Model::FanComponentModel.new(model)
+unitary_system_cc = OpenStudio::Model::CoilCoolingDXSingleSpeed.new(model)
+unitary_system_hc = OpenStudio::Model::CoilHeatingDXSingleSpeed.new(model)
+supp_unitary_system_hc = OpenStudio::Model::CoilHeatingElectric.new(model)
+# unitary_system.setControlType("SetPoint")
+unitary_system.setSupplyAirFanOperatingModeSchedule(alwaysOn)
+unitary_system.setSupplyFan(unitary_system_fan)
+unitary_system.setCoolingCoil(unitary_system_cc)
+unitary_system.setHeatingCoil(unitary_system_hc)
+unitary_system.setSupplementalHeatingCoil(supp_unitary_system_hc)
+unitary_systemAirLoopHVAC = OpenStudio::Model::AirLoopHVAC.new(model)
+unitary_systemAirLoopHVAC.setName('AirLoopHVACUnitarySystem AirLoopHVAC')
+unitary_system.addToNode(unitary_systemAirLoopHVAC.supplyOutletNode)
```

Comment:
One Fan in AirLOOPHVAC and Another fan in AirLoopHVACUnitarySystem, which are the only two supported cases for Fan:ComponentModel